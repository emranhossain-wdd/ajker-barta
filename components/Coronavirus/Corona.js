import { formatDistanceToNow } from 'date-fns';
import { useRouter } from 'next/router';
import React from 'react';

const Corona = ({ news }) => {

  const router = useRouter()
  // const { heading, description, images, publishedDate  } = news
  return (
    <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className="rounded imageEffect overflow-hidden shadow-md cursor-pointer">
      <div className="w-full overflow-hidden">
        <img className="w-full" src={news?.images?.img1} alt="Mountain" />
      </div>
      <div className="px-3 py-3">
        <div className="font-bold text-lg mb-1 leading-6 hover:text-red-600 transition-colors duration-300">{news?.heading}</div>
        <p className="text-gray-700 text-sm">
          {news?.description[0]?.slice(0, 100)}
        </p>
        <div className="mt-4">
          {news?.publishDate && <span className="inline-block bg-gray-100 rounded-full px-3 py-1 text-sm font-medium text-gray-800 mr-2 mb-2">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</span>}
        </div>
      </div>
    </div>
  );
};

export default Corona;