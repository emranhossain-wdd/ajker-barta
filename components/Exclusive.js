import { formatDistanceToNow } from 'date-fns';
import { useRouter } from 'next/router';
import React, { useState } from 'react'
import styles from '../styles/exclusive.module.css'
const Exclusive = ({ newses }) => {
  const latest = newses?.reverse()
  const router = useRouter()
  const [isLetest, setIsLetest] = useState(true)
  let count = 1
  return (
    <div data-testid='exclusiveId' className='pb-5'>
      <div className="container">
        <div className="grid lg:grid-cols-2 gap-3">
          <div>
            <button onClick={() => setIsLetest(true)} className={isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Letest</button>
            <button onClick={() => setIsLetest(false)} className={!isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Trending</button>
            {
              isLetest ? <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(0, 8).map(news => <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div> : <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(8, 16).map(news => <div onClick={() => router.push(`/news/${news._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div>
            }

          </div>
          <div>
            <h1 className={styles.exclusive}>Satireday</h1>
            <div className='grid sm:grid-cols-2 gap-3'>
              {
                latest?.slice(16, 20).map(news => <div className='cursor-pointer imageEffect' onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} key={news?._id}>
                  <div className='overflow-hidden'>
                    <img className='w-full h-32 object-cover' src={news?.images.img1} alt="" />
                  </div>
                  <h1 className='text-lg my-1 font-medium leading-5 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                  <p className='text-sm leading-4'>{news?.description[0]?.slice(0, 100)}</p>
                  <p className="px-2 mt-2 py-1 font-medium text-sm rounded-full bg-gray-100 w-fit text-blue-500">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</p>
                </div>)
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Exclusive;