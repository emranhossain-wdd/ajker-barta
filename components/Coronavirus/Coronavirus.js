import { useRouter } from "next/router";
import Corona from "./Corona";
import { IoIosArrowForward } from "react-icons/io";

const Coronavirus = ({ coronanews }) => {
  const router = useRouter();

  const latestNews = coronanews?.reverse().slice(0, 4);
  return (
    <div className="container">
      <div
        onClick={() => router.push("/coronavirus")}
        className="my-4 font-semibold text-xl cursor-pointer flex items-center gap-2"
      >
        Covid-19 <span className="text-red-500">Worldwide</span>
        <IoIosArrowForward className='text-red-600 mt-1' />
      </div>
      <div className="grid grid-cols-1 sm:grid-cols-1 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-4 gap-3">
        {latestNews?.map((news) => (
          <Corona key={news._id} news={news}></Corona>
        ))}
      </div>
    </div>
  );
};

export default Coronavirus;
