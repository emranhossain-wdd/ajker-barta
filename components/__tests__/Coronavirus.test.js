import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Coronavirus from '../Coronavirus/Coronavirus'
import { testArr } from './Exclusive.test'
import { render } from '@testing-library/react'

it('render Coronavirus component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Coronavirus />, div)
})
it('testing exlusive with props', () => {
  const { getByTestId } = render(<Coronavirus testArr={testArr} />)
  expect(getByTestId('coronaVirusId')).toBeTruthy()
})
it('maching snapshot coronavirus', () => {
  const tree = renderer.create(<Coronavirus />)
  expect(tree).toMatchSnapshot()
})
