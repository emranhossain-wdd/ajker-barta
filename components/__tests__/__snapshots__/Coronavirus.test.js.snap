// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="pb-5"
  data-testid="exclusiveId"
>
  <div
    className="container"
  >
    <div
      className="grid lg:grid-cols-2 gap-3"
    >
      <div>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white"
          onClick={[Function]}
        >
          Letest
        </button>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2"
          onClick={[Function]}
        >
          Trending
        </button>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
      <div>
        <h1
          className="exclusive"
        >
          Satireday
        </h1>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`maching snapshot coronavirus 1`] = `
<div
  className="container"
  data-testid="coronaVirusId"
>
  <div
    className="my-4 font-semibold text-xl cursor-pointer flex items-center gap-2"
    onClick={[Function]}
  >
    Covid-19 
    <span
      className="text-red-500"
    >
      Worldwide
    </span>
    <svg
      className="text-red-600 mt-1"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
      />
    </svg>
  </div>
  <div
    className="grid grid-cols-1 sm:grid-cols-1 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-4 gap-3"
  />
</div>
`;
